TestRunner.psm1

```diff
@@ -269,203 +276,379 @@ DEALLOCATE reorganizeOrRebuildCommands_cursor;
     }
 
-    Write-InfoLog "Verifying ODS API is running"
-    $url = ($configJson | ConvertFrom-Json).host
-    $webClient = New-Object System.Net.WebClient
     try {
-        [System.Net.ServicePointManager]::ServerCertificateValidationCallback = { $true }
-        $content = $webClient.DownloadString($url)
-        Write-Host $content
-    }
-    finally {
-        $webClient.Dispose()
-        [System.Net.ServicePointManager]::ServerCertificateValidationCallback = $null
-    }
-
-    $thisFolder = $PSScriptRoot
-
-    $serverMetricsBackgroundScript = {
-        param($server, [PSCredential] $credential, $getMetricsBlock, $expectedPerformanceCounters)
 
-        $commonFunctions = (Get-Command $Using:thisFolder\TestRunner.psm1).ScriptContents
+        Write-InfoLog "Writing to $testResultsPath"
 
-        $currentTestResultsFolder = $Using:testResultsPath
+        $databaseInstance = Get-ConfigValue -Config $Config -Key "PERF_DB_SERVER"
+        $databaseServer = Get-DatabaseServerName $databaseInstance
 
+        $webServer = Get-ConfigValue -Config $Config -Key "PERF_WEB_SERVER"
+        $sqlBackupFile = ""
+        $restoreDatabase = $false
         try {
-            $csvPath = "$Using:thisFolder\$currentTestResultsFolder\$Using:testType.$server.csv"
+            $restoreDatabase = [System.Convert]::ToBoolean($(Get-ConfigValue -Config $Config -Key "PERF_DB_RESTORE"))
+        } catch {
+            throw "Invalid configuration value for PERF_DB_RESTORE, should be a boolean"
+        }
+        if ($restoreDatabase) {
+            $sqlBackupFile = Get-ConfigValue -Config $Config -Key "PERF_SQL_BACKUP_FILE"
+        }
 
-            if ($server -eq 'localhost') {
-                $actualPerformanceCounters = Compare-Object (typeperf -q) $expectedPerformanceCounters -PassThru -IncludeEqual -ExcludeDifferent
+        $url = Get-ConfigValue -Config $Config -Key "PERF_API_BASEURL"
+        $key = Get-ConfigValue -Config $Config -Key "PERF_API_KEY"
+        $secret = Get-ConfigValue -Config $Config -Key "PERF_API_SECRET"
+        $clientCount = Get-ConfigValue -Config $Config -Key "PERF_CONNECTION_LIMIT" -Optional
+        if ($null -eq $clientCount) { $clientCount = 5 }
+        $contentType = Get-ConfigValue -Config $Config -Key "PERF_CONTENT_TYPE" -Optional
+        $resourceList = Get-ConfigValue -Config $Config -Key "PERF_RESOURCE_LIST" -Optional
+        $pageSize = Get-ConfigValue -Config $Config -Key "PERF_API_PAGE_SIZE" -Optional
+        $logLevel = Get-ConfigValue -Config $Config -Key "PERF_LOG_LEVEL" -Optional
+        $description =Get-ConfigValue -Config $Config -Key "PERF_DESCRIPTION" -Optional
+
+        if ($restoreDatabase) {
+            Write-InfoLog "Restoring $databaseName from $sqlBackupFile"
+
+            if (Test-IsLocalhost $databaseServer) {
+                Reset-OdsDatabase -BackupFilename $sqlBackupFile -DatabaseName $databaseInstance
```

Comment:
$databaseInstance contains the value from PERF_DB_SERVER, while restore method is expecting database name. Restore is failing

Reply:
Oh, I see what I did there - changed `DatabaseName` instead of changing `DatabaseServer`. And apparently in too much of a rush yesterday for proper testing. My apologies for that.